### DISCUSSION TITLE
ignoring (g)unicorn is it possible to bind on an IP address and port ?

### DISCUSSION THREAD

**Discussion by playgroundctl** [fastapi/fastapi#8218] [Questions] [answered]
**Description**

How can I run a FastAPI instance without using (g)unicorn ?

Is it possible to use sth like https://github.com/taoufik07/responder `run` method ?

**Additional context**
I really don't need (g)unicorn to run a http api (a prometheus client) receiving 1 request per second, hence the question :)

**Comment by wshayes** [comment]
Same way you run it in development using uvicorn directly: uvicorn main:app

**Comment by playgroundctl** [comment]
is it possible without Uvicorn ?

**Comment by wshayes** [comment, marked as answer]
FastAPI is based on ASGI (and created on top of Starlette which requires an ASGI server - see https://www.starlette.io/ <https://www.starlette.io/>). Doesn’t appear that you can run it without an ASGI server. If you are looking for something super-lightweight what about https://aiohttp.readthedocs.io/en/stable/ <https://aiohttp.readthedocs.io/en/stable/>

**Comment by playgroundctl** [comment]
thank you

**Comment by tiangolo** [comment]
Thanks for the help @wshayes ! :clap: :cake: 

@nskalis Responder is based on Starlette. And it uses Uvicorn underneath. So, it's more or less the same to use Uvicorn with FastAPI. Is just that FastAPI doesn't enforce it. You could also use Hypercorn if you need HTTP2/QUIC, Daphne, or others. But Uvicorn should do it.
